lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -341,26 +341,9 @@ abstract class BaseSlider<
   @SeparationUnit private int separationUnit = UNIT_PX;
 
   @NonNull
-  private final ViewTreeObserver.OnScrollChangedListener onScrollChangedListener =
-      () -> {
-        if (shouldAlwaysShowLabel() && isEnabled()) {
-          Rect contentViewBounds = new Rect();
-          ViewUtils.getContentView(this).getHitRect(contentViewBounds);
-          boolean isSliderVisibleOnScreen = getLocalVisibleRect(contentViewBounds);
-          for (int i = 0; i < labels.size(); i++) {
-            TooltipDrawable label = labels.get(i);
-            // Get associated value for label
-            if (i < values.size()) {
-              positionLabel(label, values.get(i));
-            }
-            if (isSliderVisibleOnScreen) {
-              ViewUtils.getContentViewOverlay(this).add(label);
-            } else {
-              ViewUtils.getContentViewOverlay(this).remove(label);
-            }
-          }
-        }
-      };
+  private final ViewTreeObserver.OnScrollChangedListener onScrollChangedListener = this::updateLabel;
+  @NonNull
+  private final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener = this::updateLabel;
```

Comment:
do we need both the global layout listener and the scroll listener or is just the global layout listener enough?

Reply:
By only the global layout listener, the slider label doesn't move while scrolling (#2869).